packages/slate/src/operations/invert.js

```diff
@@ -45,36 +45,17 @@ function invertOperation(op) {
       let inversePath = newPath
       let inverseNewPath = path
 
-      const pathLast = path.size - 1
-      const newPathLast = newPath.size - 1
+      const position = PathUtils.compare(path, newPath)
 
       // If the node's old position was a left sibling of an ancestor of
       // its new position, we need to adjust part of the path by -1.
-      if (
-        path.size < inversePath.size &&
-        path.slice(0, pathLast).every((e, i) => e == inversePath.get(i)) &&
-        path.last() < inversePath.get(pathLast)
-      ) {
-        inversePath = inversePath
-          .slice(0, pathLast)
-          .concat(inversePath.get(pathLast) - 1)
-          .concat(inversePath.slice(pathLast + 1, inversePath.size))
-      }
-
       // If the node's new position is an ancestor of the old position,
       // or a left sibling of an ancestor of its old position, we need
       // to adjust part of the path by 1.
-      if (
-        newPath.size < inverseNewPath.size &&
-        newPath
-          .slice(0, newPathLast)
-          .every((e, i) => e == inverseNewPath.get(i)) &&
-        newPath.last() <= inverseNewPath.get(newPathLast)
-      ) {
-        inverseNewPath = inverseNewPath
-          .slice(0, newPathLast)
-          .concat(inverseNewPath.get(newPathLast) + 1)
-          .concat(inverseNewPath.slice(newPathLast + 1, inverseNewPath.size))
+      if (path.size < newPath.size && position === -1) {
```

Comment:
Use the same style as the check in `element`. Some new tests were added to make sure this all still works.